subprojects/logging/src/main/java/org/gradle/api/internal/DocumentationRegistry.java

```diff
@@ -25,33 +25,63 @@
  */
 @ServiceScope(Scope.Global.class)
 public class DocumentationRegistry {
-    private final GradleVersion gradleVersion;
+    public final static String BASE_URL = "https://docs.gradle.org/" + GradleVersion.current().getVersion();
+    public static final String RECOMMENDATION = "For more %s, please refer to %s in the Gradle documentation.";
+    public static final String DSL_PROPERTY_URL_FORMAT = "%s/dsl/%s.html#%s:%s";
 
     public DocumentationRegistry() {
-        this.gradleVersion = GradleVersion.current();
     }
 
+
     /**
      * Returns the location of the documentation for the given feature, referenced by id. The location may be local or remote.
      */
     public String getDocumentationFor(String id) {
-        return String.format("https://docs.gradle.org/%s/userguide/%s.html", gradleVersion.getVersion(), id);
+        return String.format("%s/userguide/%s.html", BASE_URL, id);
     }
 
+
+    /**
+     * Returns the location of the documentation for the given feature, referenced by id and section. The location may be local or remote.
+     */
     public String getDocumentationFor(String id, String section) {
-        return String.format("https://docs.gradle.org/%s/userguide/%s.html#%s", gradleVersion.getVersion(), id, section);
+        return getDocumentationFor(id) + "#" + section;
     }
 
     public String getDslRefForProperty(Class<?> clazz, String property) {
         String className = clazz.getName();
-        return String.format("https://docs.gradle.org/%s/dsl/%s.html#%s:%s", gradleVersion.getVersion(), className, className, property);
+        return String.format(DSL_PROPERTY_URL_FORMAT, BASE_URL, className, className, property);
+    }
+
+    public String getDslRefForProperty(String className, String property) {
+        return String.format(DSL_PROPERTY_URL_FORMAT, BASE_URL, className, className, property);
     }
 
     public String getSampleIndex() {
-        return String.format("https://docs.gradle.org/%s/samples", gradleVersion.getVersion());
+        return BASE_URL + "/samples";
     }
 
     public String getSampleFor(String id) {
-        return String.format("https://docs.gradle.org/%s/samples/sample_%s.html", gradleVersion.getVersion(), id);
+        return String.format(getSampleIndex() + "/sample_%s.html", id);
+    }
+
+    public String getSampleForMessage(String id) {
+        return String.format("Learn more about Gradle by exploring Samples at %s", getSampleFor(id));
```

Comment:
The here is slightly imperative, but in the same way that “Run with --stacktrace”, or “Consult the upgrade guide” etc. is.  I don’t think anyone would find it too condescending.

You could soften it a little by saying “To learn more about” if you wanted.  You should also definitely say “exploring the Samples” or “exploring our Samples”.